test/unit/test_http_retries.py

```diff
@@ -51,6 +53,7 @@ def test_get_or_create_detector_attempts_retries(gl: Groundlight):
     )
 
 
+@flaky(max_runs=4, min_passes=1)
```

Comment:
Not quite sure whats going on here, but this one is definitely hitting our actual backend and succeeding:
```
+    ERROR test/unit/test_http_retries.py::test_get_detector_attempts_retries - openapi_client.exceptions.ApiException: (400)
+    Reason: Bad Request
+    HTTP response headers: HTTPHeaderDict({'Date': 'Thu, 15 Feb 2024 22:39:21 GMT', 'Content-Type': 'application/json', 'Content-Length': '80', 'Connection': 'keep-alive', 'Server': 'nginx/1.21.1', 'Vary': 'Accept, Origin, Cookie', 'Allow': 'GET, POST, HEAD, OPTIONS', 'X-Frame-Options': 'DENY', 'X-Content-Type-Options': 'nosniff', 'Referrer-Policy': 'same-origin', 'Cross-Origin-Opener-Policy': 'same-origin', 'X-Request-Id-Resolved': 'req_uu47ba6a13a9df4f5b9676e61e87f32476', 'Set-Cookie': 'sessionid=damdm7m4g9wz2glez8vepj7m3o0n9d6l; expires=Thu, 29 Feb 2024 22:39:21 GMT; HttpOnly; Max-Age=1209600; Path=/; SameSite=Lax'})
+    HTTP response body: ["{'__all__': ['Constraint “unique_undeleted_name_per_set” is violated.']}"]
```
Which makes me think we're somehow missing the mark with this mock. Might be worth using pdb to check if we are actually mocking the call we think we are mocking.